defaults/main.yml

```diff
@@ -65,3 +65,17 @@ kernel_settings_bootloader_cmdline: []
 # If purge is true, completely wipe out whatever the current settings
 # are and replace them with kernel_settings_parameters
 kernel_settings_purge: false
+
+# If true, the role is allowed to reboot the managed host if needed to apply
+# the changes.  If false, the role will emit a message telling the user that
+# some changes will require the managed host to be rebooted in order to be
+# applied, and will set kernel_settings_reboot_needed: true
+kernel_settings_reboot_ok: false
+
+# This is a return value from the role. If true, the role applied some change
+# which will require the managed host to be rebooted in order for the change
+# to take effect.  You can use this value to see if a reboot is needed, and
+# then reboot the machine at a covenient time in your playbook. If you would
+# rather have the role reboot the managed host when necessary, see
+# kernel_settings_reboot_ok
+kernel_settings_reboot_needed: false
```

Comment:
I think it should not be in defaults - better make it a fact. Defaults are for things that the user may want to override, but this is not meant to be overriden.